src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -23,30 +23,70 @@
 use ApiPlatform\Core\Documentation\Documentation;
 use ApiPlatform\Core\Metadata\Property\Factory\PropertyMetadataFactoryInterface;
 use ApiPlatform\Core\Metadata\Property\Factory\PropertyNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
 use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
 use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
-use ApiPlatform\Core\OpenApi\Serializer\AbstractDocumentationNormalizer;
 use ApiPlatform\Core\Operation\Factory\SubresourceOperationFactoryInterface;
 use ApiPlatform\Core\PathResolver\OperationPathResolverInterface;
 use Psr\Container\ContainerInterface;
 use Symfony\Component\PropertyInfo\Type;
 use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
 /**
- * Creates a machine readable Swagger API documentation.
+ * Generates an OpenAPI specification (formerly known as Swagger). OpenAPI v2 and v3 are supported.
  *
  * @author Amrouche Hamza <hamza.simperfit@gmail.com>
  * @author Teoh Han Hui <teohhanhui@gmail.com>
  * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Anthony GRASSIOT <antograssiot@free.fr>
  */
-final class DocumentationNormalizer extends AbstractDocumentationNormalizer
+final class DocumentationNormalizer implements NormalizerInterface, CacheableSupportsMethodInterface
 {
     use FilterLocatorTrait;
 
-    const SWAGGER_VERSION = '2.0';
+    const FORMAT = 'json';
+    const BASE_URL = 'base_url';
+    const SPEC_VERSION = 'spec_version';
+    const OPENAPI_VERSION = '3.0.2';
     const SWAGGER_DEFINITION_NAME = 'swagger_definition_name';
+    const SWAGGER_VERSION = '2.0';
+
+    /**
+     * @deprecated
+     */
     const ATTRIBUTE_NAME = 'swagger_context';
 
+    private $resourceMetadataFactory;
+    private $propertyNameCollectionFactory;
+    private $propertyMetadataFactory;
+    private $resourceClassResolver;
+    private $operationMethodResolver;
+    private $operationPathResolver;
+    private $nameConverter;
+    private $oauthEnabled;
+    private $oauthType;
+    private $oauthFlow;
+    private $oauthTokenUrl;
+    private $oauthAuthorizationUrl;
+    private $oauthScopes;
+    private $apiKeys;
+    private $subresourceOperationFactory;
+    private $paginationEnabled;
+    private $paginationPageParameterName;
+    private $clientItemsPerPage;
+    private $itemsPerPageParameterName;
+    private $paginationClientEnabled;
+    private $paginationClientEnabledParameterName;
+    private $formatsProvider;
+    private $defaultContext = [
+        self::BASE_URL => '/',
+        self::SPEC_VERSION => 2,
```

Comment:
That's the only place where default version 2 remains the default, is it for BC ?

Reply:
Yes (we will change the default in API Platform v3). WDYT?